stream-labs/streamlabs-obs
# Youtube rework

Youtube now requires to select/create an event to broadcast.

This PR includes some changes in streaming services.

# Highlights 
- 'any' is forbidden for platform's API requests
- Added more strict typechecking
- Each platform service setup stream key and update channel info in `beforeGoLive` hook
- Each platform service emit `canneInfoChanged` with updated info like `streamUrl`, `chatUrl`, `title`, `description`
- Vue components should use `StreamInfoService` for obtaining the streamInfo or listening to the `streamInfoChanged` event. `StreamInfoService` takes care of listening `canneInfoChanged` from platform services. It subscribes/unsubscribes to this event on logIn/logOut.
- `UserService` emits the 'userLogin' event even when the app starts with saved auth credentials

# Test plan
- check streaming for all platforms
- stream to new youtube event
- schedule stream to youtube and stream to the scheduled event
- check that users with manually changed streamKey are automatically migrated to the "Custom injest mode"
- check streaming works with disabled GoLive window for TW/MX
- check that disabled GoLive window setting does not apply for FB/YT
- check schedule stream for FB/YT
- check that in case of networks issues you have an error message in GoLive window and able to retry
- check that chatUrl and streamUrl are always up to date with each new YT/FB stream
- check that chatUrl and streamUrl are always up to date in Game Overlay
- check that "Use recommended settings" button works in Settings -> stream
- check streaming in the "Custom ingest mode"
- check "recording while streaming" feature

I'm going to provide  automated tests for this PR here https://github.com/stream-labs/streamlabs-obs/pull/2227
However, it's worth to go through this list before the release
